### DISCUSSION TITLE
WG-Easy always break down after several hours

### DISCUSSION THREAD

**Discussion by tsigcn** [wg-easy/wg-easy#1065] [Q&A]
I have installed wg-easy docker container and connected successfully.
But several hours later the server cannot be connected.
I have tried IOS, android, and windows client, none of them work.
It was shown to be connected on client, but acturally not connected as the wg-easy panel shown.
I have uninstalled and reinstalled wg-easy, deleted ~/.wg-easy, it does not work.
It is terrible that when I install a new server environment and install wg-easy again, it worked!
And after several hours later,  it break down again......

Wg-easy is installed on ubuntu virtual machine in azure, all necessary ports is opened.
The log of wg client is attached.
[WG_log.txt](https://github.com/wg-easy/wg-easy/files/15414037/WG_log.txt)

Any advice will be appreciated.

**Comment by gizmola** [comment]
If clients are behind NAT firewalls, it is possible you need to enable a keep alive so that the NAT firewall doesn't close the client UDP ports.
